File: Problem-056/solution.js
function powerfulDigitSum(n) {
  let max =0;
  for (let a= 1;a<n; a++){
    for (let b=1;b<n;b++){
      let power = BigInt(a)** BigInt(b);
      let sum = power.toString().split("").reduce((total,digit)=>{
        return total +Number(digit);
      },0);
      if (sum>max){
        max = sum;
      }
    }
  }
  return max;
}

File: Problem-056/solution.py
def powerful_digit_sum(n):

    maximum = 0

    for a in range(1, n):

        for b in range(1, n):

            power = a ** b

            digit_sum = sum(
                int(digit)
                for digit in str(power)
            )

            if digit_sum > maximum:
                maximum = digit_sum

    return maximum


print(powerful_digit_sum(100))
